### PR TITLE
segger: update to 3.40

### PR DESCRIPTION
**Pull request by sigvartmh**
Update to 3.40 of SystemView Target sources.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

### CODE REVIEW

**Comment by sigvartmh** [comment]
@carlescufi @nashif Couldn't add you to reviewers I don't seem to have access. I'll make a `dos2unix` conversion PR after this is merged so that it will be a lot easier to do diffs. Currently some `DOS` newlines `^M` have been removed from some strings making it a mess to diff, edit and update so my suggestion is to convert the source to UNIX so that it's standardized and better handled by git on any platform. 

Dos2unix PR: https://github.com/zephyrproject-rtos/segger/pull/16

**Comment by sigvartmh** [comment]
Also haven't made the Zephyr Update PR yet.

**Review by carlescufi — APPROVED** [review]
Looks good, thank you!

**Comment by carlescufi** [comment]
@sigvartmh did you also update this part: https://github.com/zephyrproject-rtos/segger/pull/14 ?

**Comment by sigvartmh** [comment]
@carlescufi No I'll look into adding it 👍

**Comment by carlescufi** [comment]
> @carlescufi No I'll look into adding it 👍

Thanks! It might be that it doesn't require an update, but just in case!

**Comment by sigvartmh** [comment]
@carlescufi Went through and checked if there were any changes to #14 and it was only minor commented out code and some line changes. Added it but can be removed.

**Comment by sigvartmh** [comment]
@nashif Any feedback?

**Comment by MaureenHelm** [comment]
> Also haven't made the Zephyr Update PR yet.

We need this for CI

File: Config/SEGGER_RTT_Conf.h [hunk 1]
```diff
@@ -146,22 +150,22 @@ Revision: $Rev: 21386 $
...
-    #ifdef CONFIG_MULTITHREADING
-      extern void zephyr_rtt_mutex_lock(void);
-      extern void zephyr_rtt_mutex_unlock(void);
-      #define SEGGER_RTT_LOCK() zephyr_rtt_mutex_lock()
-      #define SEGGER_RTT_UNLOCK() zephyr_rtt_mutex_unlock()
-    #else
-      extern unsigned int zephyr_rtt_irq_lock(void);
-      extern void zephyr_rtt_irq_unlock(unsigned int key);
-      #define SEGGER_RTT_LOCK() { \
-        unsigned int key = zephyr_rtt_irq_lock()
```
Comment (sigvartmh) [inline Config/SEGGER_RTT_Conf.h:159]: I need to fix some alignments

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**